io/src/main/scala/sbt/internal/io/EventMonitor.scala

```diff
@@ -127,12 +127,12 @@ private[sbt] object EventMonitor {
         recentEvents = recentEvents.filterNot(_._2.isOverdue)
         getFilesForKey(s.service.poll(delay)).foreach(maybeTrigger)
       }
-      def getFilesForKey(key: WatchKey): collection.Seq[Path] = key match {
-        case null => Nil
+      def getFilesForKey(key: WatchKey): Vector[Path] = key match {
+        case null => Vector.empty
         case k =>
-          val rawEvents = k.pollEvents
+          val rawEvents = k.pollEvents.asScala.toVector
```

Comment:
I was thinking about this line and I think that it would make sense to change it to:
```scala
val rawEvents = k.synchronized {
  val events = k.pollEvents.asScala.toVector
  k.reset()
  events
}
```
The reason is that k.reset() clears out the events for the key in the default watch service. This makes it possible to miss an event if it added between the call to pollEvents and reset (I think). That is why I put the k.reset() immediately after the call to k.pollEvents and deferred any processing of the events until after the call to reset(). I looked again at the open jdk code and all of its public methods use `synchronized(this)` so synchronization on the key should hopefully prevent us from ever missing any events.